File: cpp/Problem2125.cpp
/*
 * Problem: 2125. Number of Laser Beams in a Bank
 * URL: https://leetcode.com/problems/number-of-laser-beams-in-a-bank/
 * Difficulty: Medium
 * Tags: Math
 */

#include <algorithm>
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
	int numberOfBeams(vector<string>& bank) {
		int result = 0, prevLasterCount = 0;
		for (string &b : bank) {
			int currLasterCount = count(b.begin(), b.end(), '1');
			if (currLasterCount != 0 ){
				result += prevLasterCount * currLasterCount;
				prevLasterCount = currLasterCount;
			}
		}
		return result;
	}
};

File: python/Problem2125.py
# Problem: 2125. Number of Laser Beams in a Bank
# URL: https://leetcode.com/problems/number-of-laser-beams-in-a-bank/
# Difficulty: Medium
# Tags: Math

from typing import List

class Solution:
  def numberOfBeams(self, bank: List[str]) -> int:
    result: int = 0
    prevLasterCount: int = 0
    for b in bank:
      currLasterCount: int = b.count('1')
      result += prevLasterCount * currLasterCount
      if currLasterCount != 0:
        prevLasterCount = currLasterCount
    return result